orinoco/helpers.py

```diff
@@ -7,3 +11,17 @@ def initialize(classes: Iterable[Type[Any]]) -> List[Any]:
 
 def raise_not_provided_field(field_name: str) -> NoReturn:
     raise ValueError("Field {} has to be provided".format(field_name))
+
+
+def _extract_type(value: Union[TypeT, Annotated[TypeT, AnnotationNameT]]) -> Tuple[TypeT, Optional[AnnotationNameT]]:
+    if _is_annotated(value):
+        return value.__origin__, value.__metadata__[0]
```

Comment:
this and the check in the typeguard (which can be moved here as it's not providing the typing) can be done through typing.get_args and typing.get_origin on the passed in type hint

Reply:
Well `typing.get_origin(ano)` (-> `<class 'typing.Annotated'>`) and  `ano.__origin__`  (-> `<class 'bool'>`) returns different things...

Reply:
the get_origin gets the generic class that was parametrized (if any), get_args then lets us access what was passed to it, so I was thinking of something like this
```py
generic = typing.get_origin(ano)
if generic is None:
    return ano, None  # plain type
if generic is Annotated:
    return typing.get_args(ano)[:2]  # first two parameters to Annotated generic
return generic, None  # remove generic parameter if it was parametrized
```
the last return that removes the generic parameters is also necessary I think as parametrized generics will fail in isinstance and issubclass checks when passed for the second arg

Reply:
Nice, thanks for the tip. I was trying to make `isinstance` works with `Annotated` and forgot about using `is` for the origin :)